app/Http/Controllers/Admin/HoldAdminController.php

```diff
@@ -0,0 +1,140 @@
+     * @param HoldRequest $request
+     * @return JsonResponse
+     */
+    public function createHold(Navaid $navaid, HoldRequest $request) : JsonResponse
+    {
+        // use description if specified, otherwise default to the navaid identifier.
+        $description = $request->get('description') ?? $navaid->identifier;
+
+        if ($this->doesDescriptionAlreadyExist($navaid, $description)) {
+            return response()->json(self::DESCRIPTION_NOT_UNIQUE_RESPONSE, 409);
+        }
+
+        $hold = $navaid->holds()->create(
+            array_merge($request->validated(), compact('description'))
+        );
+
+        return response()->json(['hold_id' => $hold->id], 201);
+    }
+
+    /**
+     * Get the details of a hold associated with a Navaid.
+     *
+     * @param Navaid $navaid
+     * @param Hold $hold
+     * @return JsonResponse
+     */
+    public function getHold(Navaid $navaid, Hold $hold) : JsonResponse
+    {
+        if (! $this->checkHoldBelongsToNavaid($navaid, $hold)) {
+            return response()->json(self::HOLD_NOT_IN_NAVAID_RESPONSE, 404);
+        }
+        return response()->json(['hold' => $hold->load('restrictions')]);
+    }
+
+    /**
+     * Modify an existing hold on a specified Navaid.
+     *
+     * @param Navaid $navaid
+     * @param Hold $hold
+     * @param HoldRequest $request
+     * @return JsonResponse
+     */
+    public function modifyHold(Navaid $navaid, Hold $hold, HoldRequest $request) : JsonResponse
+    {
+        if (! $this->checkHoldBelongsToNavaid($navaid, $hold)) {
+            return response()->json(self::HOLD_NOT_IN_NAVAID_RESPONSE, 404);
+        }
+    
+        // use description if specified, otherwise default to the navaid identifier.
+        $description = $request->get('description') ?? $navaid->identifier;
+
+        $descriptionChanged = ($hold->description != $request->get('description'));
+        if ($this->doesDescriptionAlreadyExist($navaid, $description) && $descriptionChanged) {
+            return response()->json(self::DESCRIPTION_NOT_UNIQUE_RESPONSE, 409);
+        }
+
+        $hold->update($request->validated());
+ 
+        return response()->json([], 204);
+    }
+
+    /**
+     * Delete a Hold associated with a specified Navaid.
+     *
+     * @param Navaid $navaid
+     * @param Hold $hold
+     * @return JsonResponse
+     */
+    public function deleteHold(Navaid $navaid, Hold $hold) : JsonResponse
+    {
+        if (! $this->checkHoldBelongsToNavaid($navaid, $hold)) {
+            return response()->json(self::HOLD_NOT_IN_NAVAID_RESPONSE, 404);
+        }
+        $hold->delete();
+        return response()->json([], 204);
+    }
+
+    /**
+     * Checks whether the description is already used for
+     * the holds of a specified navaid.
+     *
+     * @param Navaid $navaid
+     * @param string $description
+     * @return boolean
+     */
+    private function doesDescriptionAlreadyExist(Navaid $navaid, string $description): bool
```

Comment:
Would this be better named `descriptionAlreadyExists`?

Reply:
https://github.com/VATSIM-UK/uk-controller-api/commit/4663772e05a7c8b0c7dd9185947a0d60bba0eb9a

(had a hard time naming this in the first place :D)